### ISSUE TITLE
Injecting repositories with same name in different services always gets the first one

### ISSUE THREAD

**Issue by talski** [nestjs/typeorm#958]
## Current behavior
When you have an entity with same name in different namespaces/folders, when you inject it `@InjectRepository(Category) private categoryRepository: Repository<Category>,` it always will get the first loaded class.

## Input Code
https://gitlab.com/talski/just-a-test

I could get the right instance getting the repository like this:
```ts
getConnection().getRepository(Category)
```
## Expected behavior
Should get the right instance using the @InjectRepository

## Environment
```
Nest version: v8.0.5
 
For Tooling issues:
- Node version: v15.10.0
- Platform: Linux
```

**Comment by jmcdo29** [comment]
In your reproduction, can you include a `docker-compose.yml` so that we don't have to set that up ourselves? Thanks.

**Comment by talski** [comment]
> In your reproduction, can you include a `docker-compose.yml` so that we don't have to set that up ourselves? Thanks.

done

**Comment by jmcdo29** [comment]
Okay, this does reproduce, and I think it comes from the entity-metadata storage that Nest is using. It only takes into account the entity, and I'm not sure if it changes unless the class name changes. We can probably employ a similar pattern as the Nest v8 update to the provider resolution, to use a class reference instead. 

I don't often see this kind of setup, so it definitely seems like an edge case that's unaccounted for. @kamilmysliwiec what do you think?

**Comment by kamilmysliwiec** [comment]
This is caused by the Nest dynamic modules caching/metadata serialization issue.

Fixed in this commit https://github.com/nestjs/typeorm/commit/fed70fb04fb92e5f659d07f9c83a47ca7b8b3d1a. Published as 8.0.2
